service/controller/app/v1/resource/chart/create_test.go

```diff
@@ -0,0 +1,263 @@
+package chart
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func TestResource_GetCurrentState(t *testing.T) {
+
+	tests := []struct {
+		name          string
+		obj           *v1alpha1.App
+		returnChart   *v1alpha1.Chart
+		expectedChart *v1alpha1.Chart
+		errorMatcher  func(error) bool
+	}{
+		{
+			name: "case 0: chart already created",
+			obj: &v1alpha1.App{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      "my-cool-prometheus",
+					Namespace: "default",
+					Labels: map[string]string{
```

Comment:
IMO you could remove the labels and annotations here. As they aren't needed for the test. Just to make it easier to read.